### PR TITLE
Headline: clang-format struck again

### LINKED ISSUE #6752: broken doxygen from clang-format comment wrapping
There are several instances where clang-format messes up the doxygen formatting. For example:
http://www.dealii.org/developer/doxygen/deal.II/classScaLAPACKMatrix.html#a84e89c6cfa606aaf5d39e981de7ae378

How do we fix this? What is the best way to disable the comment wrapping?

### PR DESCRIPTION
**Pull request by tamiko**
Fixes #6752

### CODE REVIEW

**Comment by tamiko** [comment]
@tjhei Here you go.

This is what I got by
 - tracking the differences between the old astyle formatting and the first application of clang-format
 - forward porting these changes to current master
 - running `wrap_comments.py` for good measure on our `include` directory.

**Comment by tamiko** [comment]
I removed the `wrap_comments.py` run again. It garbles up latex in \f blocks!

File: include/deal.II/lac/scalapack.h [hunk 1]
```diff
@@ -252,12 +252,10 @@ class ScaLAPACKMatrix : protected TransposeTable<NumberType>
...
-   * | transpose_B |          Block Sizes         |                    Operation
-   * | | :---------: | :--------------------------: |
-   * :-------------------------------------------: | |   false     | $MB_A=MB_B$
-   * <br> $NB_A=NB_B$ |  $\mathbf{A} = a \mathbf{A} + b \mathbf{B}$   | |   true
-   * | $MB_A=NB_B$ <br> $NB_A=MB_B$ | $\mathbf{A} = a \mathbf{A} + b
-   * \mathbf{B}^T$  |
+   * | transpose_B |          Block Sizes         |                    Operation                  |
+   * | :---------: | :--------------------------: | :-------------------------------------------: |
+   * |   false     | $MB_A=MB_B$ <br> $NB_A=NB_B$ |  $\mathbf{A} = a \mathbf{A} + b \mathbf{B}$   |
+   * |   true      | $MB_A=NB_B$ <br> $NB_A=MB_B$ | $\mathbf{A} = a \mathbf{A} + b \mathbf{B}^T$  |
```
Comment (tamiko) [inline include/deal.II/lac/scalapack.h]: @bangerth @tjhei Everything that looks like a table (containing ` | `) will not be touched.

File: source/grid/tria.cc [hunk 2]
```diff
@@ -14511,8 +14511,13 @@ Triangulation<dim, spacedim>::prepare_coarsening_and_refinement()
...
-                                    //                          | | | | | | | |
-                                    //                          | | | | / | |/
+                                    //                          |               |
+                                    //                          |               |
+                                    //                          |               |
+                                    //                          |               |
+                                    //                          |               |
+                                    //                          |               | /
+                                    //                          |               |/
                                     //                          *---------------*
```
Comment (tamiko) [inline source/grid/tria.cc]: Also fixes ascii art...

Comment (tjhei) [reply source/grid/tria.cc]: yay!

**Comment by masterleinad** [comment]
Just to be on the safe side:

/run-testsuite

**Comment by tamiko** [comment]
@masterleinad Passed all tests - I had to rebase again (?!).

**Pull request merged**